josh-core/src/filter/mod.rs

```diff
@@ -482,20 +484,23 @@ fn apply_to_commit2(
 
             let id = commit.id();
 
-            if let Some(startfilter) = filters.get(&id) {
-                let mut f2 = filters.clone();
-                f2.remove(&id);
-                f2.insert(git2::Oid::zero(), *startfilter);
-                let op = if f2.len() == 1 {
-                    to_op(*startfilter)
-                } else {
-                    Op::Rev(f2)
-                };
-                if let Some(start) = apply_to_commit2(&op, commit, transaction)? {
-                    transaction.insert(filter, id, start, true);
-                    return Ok(Some(start));
-                } else {
-                    return Ok(None);
+            for (&filter_tip, startfilter) in filters.iter() {
```

Comment:
This iterates the filters in unspecified order, as HashMap does not have stable iteration order. This means if the parent-sets of the commits given in a `rev` overlap, behavior would be non-deterministic...

Reply:
`filters` is already a `BTreeMap` for exactly that reason.
It's probably slower, but determinism is more important.
Looking at the usage we could change it to an ordered `Vec` though 🤔

Reply:
Alternatively, maybe it should error if two of the parent-sets overlap?